core/services/flux_monitor.go

```diff
@@ -0,0 +1,342 @@
+package services
+
+import (
+	"chainlink/core/eth"
+	"chainlink/core/logger"
+	"chainlink/core/store"
+	"chainlink/core/store/models"
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/shopspring/decimal"
+	"go.uber.org/multierr"
+)
+
+// FluxMonitor is the interface encapsulating all functionality
+// needed to listen to price deviations and new round requests.
+type FluxMonitor interface {
+	store.HeadTrackable // (Dis)Connect methods handle initial boot and intermittent connectivity.
+	AddJob(models.JobSpec) error
+	RemoveJob(*models.ID)
+}
+
+type concreteFluxMonitor struct {
+	store          *store.Store
+	runManager     RunManager
+	checkers       map[string][]DeviationChecker
+	checkerFactory DeviationCheckerFactory
+	adds           chan addEntry
+	removes        chan *models.ID
+	ctx            context.Context
+	cancel         context.CancelFunc
+}
+
+type addEntry struct {
+	job   *models.JobSpec
+	rchan chan error
+}
+
+// NewFluxMonitor creates a service that manages a collection of DeviationCheckers,
+// one per initiator of type InitiatorFluxMonitor for added jobs.
+func NewFluxMonitor(store *store.Store, runManager RunManager) FluxMonitor {
+	return &concreteFluxMonitor{
+		store:          store,
+		runManager:     runManager,
+		checkerFactory: pollingDeviationCheckerFactory{},
+	}
+}
```

Comment:
The [best argument](https://peter.bourgon.org/go-for-industrial-programming/#how-much-interface-do-i-need)  for this is best summed up by Peter Bourgon

Reply:
👍

Reply:
The reason it wasn't exported is because of this `golint` message:

> exported func NewFluxMonitor returns unexported type *services.concreteFluxMonitor, which can be annoying to use (golint)

Leaving it as is for now. I prefer keeping things unexported if possible to reduce the surface area of the package.

Reply:
Ok I also noticed the same pattern is used throughout the package for the other services.  Is the interface exported in support of the tests?